### PR TITLE
Feature/update play new

### PR DESCRIPTION
**Pull request by RobotJente**

### CODE REVIEW

File: include/roboteam_ai/stp/Play.hpp [hunk 1]
```diff
@@ -43,6 +43,8 @@ class Play {
      */
     [[nodiscard]] virtual Status update() noexcept;
 
+    [[nodiscard]] virtual void calculateInfoForPlay() noexcept = 0;
```
Comment (timovdk) [inline include/roboteam_ai/stp/Play.hpp]: void return type doesn't need nodiscard
```suggestion
    virtual void calculateInfoForPlay() noexcept = 0;
```

File: include/roboteam_ai/stp/new_plays/TestPlay.h [hunk 2]
```diff
@@ -38,6 +38,8 @@ class TestPlay : public Play {
      * Assigns robots to roles of this play
      */
     void assignRoles() noexcept;
+
+    [[nodiscard]] virtual void calculateInfoForPlay() noexcept override;
```
Comment (RobotJente) [inline include/roboteam_ai/stp/new_plays/TestPlay.h]:
```suggestion
    [[nodiscard]] void calculateInfoForPlay() noexcept override;
```

File: test/StpTests/PlayCheckerTests.cpp [hunk 3]
```diff
@@ -18,6 +18,10 @@ class AlwaysValid : public rtt::ai::stp::Play {
 
     void assignRoles() noexcept override {
     }
+    [[nodiscard]] virtual void calculateInfoForPlay() noexcept {
+
```
Comment (RobotJente) [inline test/StpTests/PlayCheckerTests.cpp]:
```suggestion
```

File: include/roboteam_ai/stp/new_plays/TestPlay.h (hunk 2 above)
Comment (timovdk) [inline include/roboteam_ai/stp/new_plays/TestPlay.h]:
```suggestion
    virtual void calculateInfoForPlay() noexcept override;
```

File: test/StpTests/PlayCheckerTests.cpp [hunk 4]
```diff
@@ -31,6 +35,10 @@ class AlwaysFalse : public rtt::ai::stp::Play {
 
     void assignRoles() noexcept override {
     }
+
+    [[nodiscard]] virtual void calculateInfoForPlay() noexcept {
+
```
Comment (RobotJente) [inline test/StpTests/PlayCheckerTests.cpp]:
```suggestion
```

File: include/roboteam_ai/stp/new_plays/TestPlay.h [hunk 5]
```diff
@@ -38,6 +38,8 @@ class TestPlay : public Play {
      * Assigns robots to roles of this play
      */
     void assignRoles() noexcept;
+
+    [[nodiscard]] void calculateInfoForPlay() noexcept override;
```
Comment (timovdk) [inline include/roboteam_ai/stp/new_plays/TestPlay.h]:
```suggestion
    void calculateInfoForPlay() noexcept override;
```

File: test/StpTests/PlayCheckerTests.cpp [hunk 6]
```diff
@@ -18,6 +18,9 @@ class AlwaysValid : public rtt::ai::stp::Play {
 
     void assignRoles() noexcept override {
     }
+    [[nodiscard]] virtual void calculateInfoForPlay() noexcept {
```
Comment (timovdk) [inline test/StpTests/PlayCheckerTests.cpp]:
```suggestion
    virtual void calculateInfoForPlay() noexcept {
```

File: test/StpTests/PlayCheckerTests.cpp [hunk 7]
```diff
@@ -31,6 +34,9 @@ class AlwaysFalse : public rtt::ai::stp::Play {
 
     void assignRoles() noexcept override {
     }
+
+    [[nodiscard]] virtual void calculateInfoForPlay() noexcept {
```
Comment (timovdk) [inline test/StpTests/PlayCheckerTests.cpp]:
```suggestion
    virtual void calculateInfoForPlay() noexcept {
```

File: test/StpTests/PlayCheckerTests.cpp [hunk 8]
```diff
@@ -18,6 +18,9 @@ class AlwaysValid : public rtt::ai::stp::Play {
 
     void assignRoles() noexcept override {
     }
+    virtual void calculateInfoForPlay() noexcept {
+    }
+
```
Comment (timovdk) [inline test/StpTests/PlayCheckerTests.cpp]:
```suggestion
```

**Review by timovdk — APPROVED** [review]

**Review by timovdk — CHANGES_REQUESTED** [review]
Travis does not pass

**Review by timovdk — APPROVED** [review]

**Pull request merged**